carla/recourse_methods/catalog/growing_spheres/library/gs_counterfactuals.py

```diff
@@ -137,7 +139,7 @@ def growing_spheres_search(
                 (candidate_counterfactuals.values - instance_replicated)
             ).sum(axis=1)
         else:
-            print("Distance not defined yet")
+            log.info("Distance not defined yet")
```

Comment:
Potentialy a `log.error` or a `log.warning`?

Reply:
I would go one step further and raise a `ValueError` here, since this could cause an error on runtime.